lib/readline.js

```diff
@@ -326,6 +341,28 @@ Interface.prototype.question = function(query, cb) {
   }
 };
 
+Interface.prototype.question[promisify.custom] = function(query, options) {
+  options = typeof options === 'object' ? options : {};
+
+  return new Promise((resolve, reject) => {
+    this.question(query, options, resolve);
+
+    if (options.signal) {
+      options.signal.addEventListener('abort', () => {
+        reject(new ERR_READLINE_CANCEL());
+      }, { once: true });
+    }
+  });
+};
+
+Interface.prototype._questionCancel = function() {
```

Comment:
we should probably not attach these stuff into the prototype anymore. This is something, we had discussion last time in changing the readline/repo codebase. Thoughts cc @BridgeAR

Reply:
I can change it but may need a pointer to how you want it. Changing the whole readline to not add stuff to the prototype seems like out of scope for this pull-request to me. If you tell me how you want it I could do another pull-request to change that if you wish.